lighthouse/helpers/plates.py

```diff
@@ -470,21 +495,174 @@ def create_cherrypicked_post_body(barcode: str, samples: List[Dict[str, Any]]) -
+                "friendly_name": control_friendly_name(sample),
+                "uuid": sample["uuid"],
+            }
+        else:
+            subject = {
+                "role_type": "sample",
+                "subject_type": "sample",
+                "friendly_name": sample_friendly_name(sample),
+                "uuid": sample["uuid"],
+            }
+        subjects.append(subject)
+    return subjects
+
+
+def sample_friendly_name(sample):
+    name = "__".join(
+        [sample["sample_description"], sample["name"], sample["lab_id"], sample["phenotype"]]
+    )
+    return name
+
+
+def control_friendly_name(sample):
+    return f"{sample['supplier_name']}"
+
+
+# def destination_labware_subject(barcode):
+#     subject = {
+#         "role_type": "cherrypicking_destination_labware",
+#         "subject_type": "plate",
+#         "friendly_name": barcode,
+#     }
+#     return subject
+
+
+def source_plate_subjects(plate_id_mappings):
+    subjects = []
+    for mapping in plate_id_mappings:
+        subject = {
+            "role_type": "cherrypicking_source_labware",
+            "subject_type": "plate",
+            "friendly_name": mapping["barcode"],
+            "uuid": mapping["uuid"],
+        }
+        subjects.append(subject)
+    return subjects
+
+
+def robot_subject(robot_serial_number):
+    try:
+        robot_mapping = app.config["BECKMAN_ROBOTS"][robot_serial_number]
+    except KeyError:
+        logger.error("Unable to find events information for robot:" + robot_serial_number)
+        raise
+    try:
+        robot_friendly_name = robot_mapping["name"]
+    except KeyError:
+        logger.error("Unable to find friendly name for robot: " + robot_serial_number)
+        raise
+
+    try:
+        robot_uuid = robot_mapping["uuid"]
+    except KeyError:
+        logger.error("Unable to find UUID for robot: " + robot_serial_number)
+        raise
+
+    subject = {
+        "role_type": "robot",
+        "subject_type": "robot",
+        "friendly_name": robot_friendly_name,
+        "uuid": robot_uuid,
+    }
+
+    return subject
+
+
+def get_source_plate_id_mappings(samples):
+    barcodes = get_unique_plate_barcodes(samples)
+    source_plate_documents = find_source_plates(query_for_source_plate_uuids(barcodes))
+
+    source_plate_uuids = []
+    for plate in source_plate_documents:
+        mapping = {
+            "barcode": plate[FIELD_BARCODE],
+            "uuid": plate[FIELD_LH_SOURCE_PLATE_UUID],
+        }
+        source_plate_uuids.append(mapping)
+
+    return source_plate_uuids
+
+
+def find_source_plates(query: Dict[str, Any]) -> Optional[List[Dict[str, Any]]]:
+    if query is None:
+        return None
+
+    source_plates = app.data.driver.db.source_plates
+
+    source_plate_documents = list(source_plates.find(query))
+
+    logger.info(f"Found {len(source_plate_documents)} source plates")
+
+    return source_plate_documents
+
+
+def get_unique_plate_barcodes(samples):
```

Comment:
```
barcodes = [sample[FIELD_PLATE_BARCODE] for sample in samples]
return list(set(barcodes))
```
Using both the generator format and converting to a set (then back to a list), seems a more succinct way to do this, as a set's elements are unique 😄

Reply:
Maybe even better (quicker) to use a set from the beginning?
```
barcodes = set()
for sample in samples:
    barcodes.add(sample[FIELD_PLATE_BARCODE])
return list(barcodes)
```

Reply:
Yep that looks better, I've updated